### PR TITLE
Address issues found by zizmor

### LINKED ISSUE #107: Address issues discovered by zizmor
```
🌈 completed hugo.yaml
warning[artipacked]:** credential persistence through GitHub Actions artifacts
  --> .github/workflows/hugo.yaml:43:9
   |
43 |         - name: Checkout
   |  _________-
44 | |         uses: actions/checkout@v3
45 | |         with:
46 | |           submodules: recursive
47 | |           fetch-depth: 0
   | |________________________- does not set persist-credentials: false
   |
   = note: audit confidence → Low

error[excessive-permissions]: overly broad workflow or job-level permissions
  --> .github/workflows/hugo.yaml:14:1
   |
14 | / permissions:
15 | |   contents: read
...  |
19 | | # Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
20 | | # However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
   | |_______________________________________________________________________________________________________^ id-token: write is overly broad at the workflow level
   |
   = note: audit confidence → High

error[excessive-permissions]: overly broad workflow or job-level permissions
  --> .github/workflows/hugo.yaml:14:1
   |
14 | / permissions:
15 | |   contents: read
...  |
19 | | # Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
20 | | # However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
   | |_______________________________________________________________________________________________________^ pages: write is overly broad at the workflow level
   |
   = note: audit confidence → High

info[template-injection]: code injection via template expansion
  --> .github/workflows/hugo.yaml:37:9
   |
37 |         - name: Install Hugo CLI
   |           ---------------------- info: this step
38 |           run: |
   |  _________-
39 | |           wget -O ${{ runner.temp }}/hugo.deb https://github.com/gohugoio/hugo/releases/download/v${HUGO_VERSION}/hugo_extended_${HU...
40 | |           && sudo dpkg -i ${{ runner.temp }}/hugo.deb          
   | |_____________________________________________________- info: runner.temp may expand into attacker-controllable code
   |
   = note: audit confidence → Low

info[template-injection]: code injection via template expansion
  --> .github/workflows/hugo.yaml:37:9
   |
37 |         - name: Install Hugo CLI
   |           ---------------------- info: this step
38 |           run: |
   |  _________-
39 | |           wget -O ${{ runner.temp }}/hugo.deb https://github.com/gohugoio/hugo/releases/download/v${HUGO_VERSION}/hugo_extended_${HU...
40 | |           && sudo dpkg -i ${{ runner.temp }}/hugo.deb          
   | |_____________________________________________________- info: runner.temp may expand into attacker-controllable code
   |
   = note: audit confidence → Low

info[template-injection]: code injection via template expansion
  --> .github/workflows/hugo.yaml:53:9
   |
53 |         - name: Build with Hugo
   |           --------------------- info: this step
54 |           env:
...   
57 |             HUGO_ENV: production
58 |           run: |
   |  _________-
59 | |           hugo \
60 | |             --gc \
61 | |             --minify \
62 | |             --baseURL "${{ steps.pages.outputs.base_url }}/"          
   | |____________________________________________________________- info: steps.pages.outputs.base_url may expand into attacker-controllable code
   |
   = note: audit confidence → Low

6 findings (0 ignored): 0 unknown, 3 informational, 0 low, 1 medium, 2 high
```

### PR DESCRIPTION
**Pull request by funnelfiasco**
Fixes #107

I chose to ignore `runner.temp` because I don't think it's an actual problem the way we use it. Upstream [agrees but not confidently enough to have a built-in ignore](https://github.com/woodruffw/zizmor/pull/128#issuecomment-2460454530).

### CODE REVIEW

File: zizmor.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Configuration file for zizmor GitHub action checker
+rules:
+  template-injection:
+    ignore:
+      # Ignore the use of `runner.temp`, as we believe it to be safe
+      - hugo.yaml:34:9
+      - hugo.yaml:51:9
```
Comment (mihaimaruseac) [inline zizmor.yml:7]: Should this have a `\n` at the end of the file?

File: .github/workflows/hugo.yaml [hunk 2]
```diff
@@ -10,11 +10,8 @@ on:
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
-# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
 permissions:
   contents: read
```
Comment (mihaimaruseac) [inline .github/workflows/hugo.yaml:14]: I think these are the default, so `permissions: {}` also works (https://github.com/sigstore/model-transparency/blob/bca8a012a6c29f0f9088f1b46f7150a597fe4f77/.github/workflows/release.yml#L22 works for sure as we have run it before on release)

**Review by mihaimaruseac — APPROVED** [review]
Minor nits

File: zizmor.yml (hunk 1 above)
Comment (funnelfiasco) [reply zizmor.yml:7]: zizmor does not require it, but I can add one for tidyness.

File: .github/workflows/hugo.yaml (hunk 2 above)
Comment (funnelfiasco) [reply .github/workflows/hugo.yaml:14]: Makes sense. I just left it because it had been there previously and I was trying to avoid breaking anything :-)

**Pull request merged**